src/impl/format.ts

```diff
@@ -88,76 +98,151 @@ export function format(documentText: string, range: Range | undefined, options:
 			addEdit(' ', firstTokenEnd, commentTokenStart);
 			firstTokenEnd = scanner.getTokenOffset() + scanner.getTokenLength() + formatTextStart;
 			needsLineBreak = secondToken === SyntaxKind.LineCommentTrivia;
-			replaceContent = needsLineBreak ? newLineAndIndent() : '';
+			replaceContent = needsLineBreak ? newLinesAndIndent() : '';
 			secondToken = scanNext();
 		}
 
 		if (secondToken === SyntaxKind.CloseBraceToken) {
-			if (firstToken !== SyntaxKind.OpenBraceToken) {
+			if (options.keepLines) {
+				if (firstToken !== SyntaxKind.OpenBraceToken) { indentLevel--; };
```

Comment:
I would move the indentLevel update before the `if (options.keepLines) {` as it is common to both variants